### ISSUE TITLE
question about AUROC_sp is 100 when test

### ISSUE THREAD

**Issue by WenDongyp** [zhangzjn/ADer#46]
I found AUROC_sp in the code that directly uses the pr_sp output from the model to calculate the image-level metric, but this metric is 100 on MVTec-AD for all classes, which is incredibly high. I wonder what the reason is?
<img width="1254" alt="test_result" src="https://github.com/user-attachments/assets/6bb7755a-f954-4829-9f13-07761eef5cf7">
<img width="267" alt="auroc_sp" src="https://github.com/user-attachments/assets/7762413b-8ca6-4461-b283-bd9ab01e49d4">
<img width="328" alt="pr_sp" src="https://github.com/user-attachments/assets/52418387-1c54-4b51-be45-40112687ebde">

**Comment by zhangzjn** [comment]
This metric objectively reaches saturation on MVTec AD, but performs poorly in practical testing. Therefore, we need new metrics that are closer to real-world applications and multiple metrics to evaluate the effectiveness of different methods.

**Comment by uobinxiao** [comment]
> This metric objectively reaches saturation on MVTec AD, but performs poorly in practical testing. Therefore, we need new metrics that are closer to real-world applications and multiple metrics to evaluate the effectiveness of different methods.

No, It seems that it is a bug.  You are getting the prediction through pr_sp = results['anomalys'][idxes], but it seems that this is ground truth. You can check the util/metrics.py
